testing/test_capture.py

```diff
@@ -1490,3 +1490,36 @@ def test_fails():
     result_with_capture.stdout.fnmatch_lines(
         ["E * TypeError: write() argument must be str, not bytes"]
     )
+
+
+def test_logging_while_collecting_skipif(testdir):
```

Comment:
The `skipif` is mostly a red herring -- please remove from the test name and replace the test file with this:

```py
import logging

logging.warning("during collection")

def test_logging():
    logging.warning("during call")
    assert False
```

Reply:
Thanks for the simpler MWE!